eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,225 @@
+            }
+        }
+    }
+
+    /**
+     * Check if the Logger is static instance, log as error if static instance logger,
+     * @param varDefAST the VARIABLE_DEF node
+     */
+    private void isNonStaticLoggerInstance(DetailAST varDefAST) {
+
+        String type = varDefAST.findFirstToken(TokenTypes.TYPE).getFirstChild().getText();
+        if (type.equals(LOGGER_CLASS)) {
+            if(varDefAST.findFirstToken(TokenTypes.MODIFIERS).branchContains(TokenTypes.LITERAL_STATIC)) {
+                log(varDefAST.getLineNo(), STATIC_LOGGER_ERR);
+            }
+        }
+    }
+
+    /**
+     * Always guard with conditional block whenever a log function called.
+     * @param methodCallAST The METHOD_CALL node
+     */
+    private void isGuardConditionChecked(DetailAST methodCallAST) {
+        DetailAST loggerName = methodCallAST.findFirstToken(TokenTypes.DOT).getFirstChild();
+        // only check for logger instance, skip checking otherwise
+        if (!loggerSet.contains(loggerName.getText())) {
+            return;
+        }
+
+        DetailAST methodName = methodCallAST.findFirstToken(TokenTypes.DOT).getLastChild();
+        String methodNameStr = methodName.getText();
+        if (isLogLevelMatched(methodCallAST, methodNameStr)) {
+            switch (methodNameStr) {
+                case FATAL: // log4j only
+                    log(methodCallAST.getLineNo(), String.format(LOGGER_LEVEL_ERR, methodNameStr, IS_FATAL_ENABLED));
+                    break;
+                case ERROR:
+                    log(methodCallAST.getLineNo(), String.format(LOGGER_LEVEL_ERR, methodNameStr, IS_ERROR_ENABLED));
+                    break;
+                case WARN:
+                    log(methodCallAST.getLineNo(), String.format(LOGGER_LEVEL_ERR, methodNameStr, IS_WARN_ENABLED));
+                    break;
+                case INFO:
+                    log(methodCallAST.getLineNo(), String.format(LOGGER_LEVEL_ERR, methodNameStr, IS_INFO_ENABLED));
+                    break;
+                case DEBUG:
+                    log(methodCallAST.getLineNo(), String.format(LOGGER_LEVEL_ERR, methodNameStr, IS_DEBUG_ENABLED));
+                    break;
+                case TRACE:
+                    log(methodCallAST.getLineNo(), String.format(LOGGER_LEVEL_ERR, methodNameStr, IS_TRACE_ENABLED));
+                    break;
+            }
+        }
+    }
+    
+    /**
+     * @param methodCallRoot METHOD_CALL node which contains logging method call, such as logger.debug()
+     * @param loggingLevelName  logging level method name, such as debug, error, info, etc.
+     * @return return true if the logging level match with guard condition block check, otherwise, return false.
+     */
+    private boolean isLogLevelMatched(DetailAST methodCallRoot, String loggingLevelName) {
+        if (methodCallRoot == null || methodCallRoot.getParent() == null) {
```

Comment:
You call this method regardless of if the `loggingLevelName` is in the "error", "info", "warn" set. The only check from the calling method above is that the variable has a name in the `loggingSet`. It's possible that the user is calling `logger.isEnabled()` and it'll flow into this method.